### PR TITLE
generate openapi spec api-doc

### PR DESCRIPTION
**Pull request by ydayagi**
FLPATH-1774
https://issues.redhat.com/browse/FLPATH-1774

### CODE REVIEW

**Comment by batzionb** [comment]
@ydayagi 
Is this the postinstall script we talked about?

File: workspaces/orchestrator/plugins/orchestrator-common/src/openapi/openapi.yaml [hunk 1]
```diff
@@ -357,7 +357,7 @@ components:
         workflowId:
           type: string
           description: Workflow unique identifier
-          minLength: 1
+          minLength: 2
```
Comment (batzionb) [inline workspaces/orchestrator/plugins/orchestrator-common/src/openapi/openapi.yaml:360]: is there a reason for this change here?

File: workspaces/orchestrator/scripts/postinstall [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/bin/bash
+mydir=`pwd`
+cd ../../ && yarn install
```
Comment (batzionb) [inline workspaces/orchestrator/scripts/postinstall:3]: this shouldn't run yarn install
the developer/Ci runs yarn install
and npm should run the script after yarn install is done

File: workspaces/orchestrator/scripts/postinstall [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+cd $mydir
+cd plugins/orchestrator-common
+yarn openapi:check
```
Comment (batzionb) [inline workspaces/orchestrator/scripts/postinstall:6]: I thought the idea was to remove the whole generated code and run yarn openapi:generate
Not check it here

File: workspaces/orchestrator/scripts/postinstall (hunk 2 above)
Comment (ydayagi) [reply workspaces/orchestrator/scripts/postinstall:3]: this is taken from the original line in package.json

File: workspaces/orchestrator/plugins/orchestrator-common/src/openapi/openapi.yaml (hunk 1 above)
Comment (ydayagi) [reply workspaces/orchestrator/plugins/orchestrator-common/src/openapi/openapi.yaml:360]: it is just a test. I am not merging yet

**Comment by batzionb** [comment]
@ydayagi 
A cleaner way to do the postinstall is to add to 
workspaces/orchestrator/plugins/orchestrator-common/package.json to scripts field this:
 "postinstall": "openapi:generate"
 
 and then you don't need the script

File: workspaces/orchestrator/scripts/postinstall (hunk 3 above)
Comment (batzionb) [reply workspaces/orchestrator/scripts/postinstall:6]: ok

File: workspaces/orchestrator/plugins/orchestrator-common/src/openapi/openapi.yaml (hunk 1 above)
Comment (batzionb) [reply workspaces/orchestrator/plugins/orchestrator-common/src/openapi/openapi.yaml]: great

File: workspaces/orchestrator/plugins/orchestrator-common/src/openapi/api-doc-template.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+metadata:
+  name: orchestrator-backend
+  description: The Orchestrator backend plugin API
+  tags:
+    - orchestrator
+spec:
+  type: openapi
+  lifecycle: experimental
+  owner: development
+  definition: |
```
Comment (batzionb) [inline workspaces/orchestrator/plugins/orchestrator-common/src/openapi/api-doc-template.yaml:12]: what is the definition supposed to be?

Comment (ydayagi) [reply workspaces/orchestrator/plugins/orchestrator-common/src/openapi/api-doc-template.yaml:12]: this is  a template file. the script generates the doc file from it. it appends the openapi spec

**Comment by batzionb** [comment]
@ydayagi 
I checked out the PR and saw how the orchestrator-api.yaml is generated
Can you add the final file to the source code? Same as all generated files are in the source code
Otherwise after running openapi:generated it appears as a change in the repo (it's not part of .gitignore)

**Comment by ydayagi** [comment]
> @ydayagi I checked out the PR and saw how the orchestrator-api.yaml is generated Can you add the final file to the source code? Same as all generated files are in the source code Otherwise after running openapi:generated it appears as a change in the repo (it's not part of .gitignore)

I forgot to commit it

File: workspaces/orchestrator/scripts/postinstall (hunk 2 above)
Comment (batzionb) [reply workspaces/orchestrator/scripts/postinstall:3]: got it. thanks

**Review by batzionb — APPROVED** [review]

File: workspaces/orchestrator/.changeset/polite-chicken-applaud.md
Comment (gciavarrini) [inline workspaces/orchestrator/.changeset/polite-chicken-applaud.md:1]: @ydayagi why this name?

Comment (ydayagi) [reply workspaces/orchestrator/.changeset/polite-chicken-applaud.md:1]: it is autogenerated

Comment (gciavarrini) [reply workspaces/orchestrator/.changeset/polite-chicken-applaud.md:1]: but all the others are like this one https://github.com/redhat-developer/rhdh-plugins/tree/main/workspaces/bulk-import/.changeset 
do you still think it's correct?

Comment (batzionb) [reply workspaces/orchestrator/.changeset/polite-chicken-applaud.md:1]: @gciavarrini 
there are the temporary changeset names that are auto generated,
Once the release PR gets merged, these are deleted and there's only one changeset file

**Pull request merged**